cider/exec/plan/parser/SubstraitToAnalyzerExpr.cpp

```diff
@@ -961,22 +969,22 @@ std::shared_ptr<Analyzer::Expr> Substrait2AnalyzerExprConverter::toAnalyzerExpr(
     const std::unordered_map<int, std::string> function_map,
     std::shared_ptr<std::unordered_map<int, std::shared_ptr<Analyzer::Expr>>>
         expr_map_ptr) {
-  // For date type, return an Analyzer::Constant expr directly, will not do cast.
+  // TODO: (spevenhe) this branch should only cover cast_string_to_date case.
+  // encapsulate and support nested function like cast(cast(xx as String) AS Date).
+  // CAST(SUBSTRING(col_string, 0, 5) AS DATE
+  // Now only supports cast(col_string AS Date)
   if (s_cast_expr.type().kind_case() == substrait::Type::kDate) {
     SQLTypeInfo sqlTypeInfo = getSQLTypeInfo(s_cast_expr.type());
-    // Default is none encoding
-    // sqlTypeInfo.set_compression(EncodingType::kENCODING_NONE);
-    Datum v;
-    // TODO: currently only support a string date representation. Seems it's possible to
-    // use a int32/int64 to represent under substrait protocol
     if (!s_cast_expr.input().literal().fixed_char().empty()) {
+      // Default is none encoding
+      // sqlTypeInfo.set_compression(EncodingType::kENCODING_NONE);
+      Datum v;
+      // TODO: currently only support a string date representation. Seems it's possible to
```

Comment:
This comment should also be updated? for literal, should we support cast literal timestamp to Date?

Reply:
No code change. Currently, Cast literal timestamp to date will go through normal cast_expr branch since cider internal method has implemented the type conversion of time to date.
https://github.com/intel/BDTK/blob/dcb64358c35f27d9f6e6d839a3e9a143315f4049/cider/exec/plan/parser/SubstraitToAnalyzerExpr.cpp#L1038-L1043